### PR TITLE
Apply some clippy suggestions

### PR DESCRIPTION
**Pull request by clemenswasser**

### CODE REVIEW

File: crates/hir_def/src/attr.rs [hunk 1]
```diff
@@ -675,10 +675,8 @@ impl Attr {
                 _ => lit.syntax().first_token()?.text().trim_matches('"').into(),
             };
             Some(Interned::new(AttrInput::Literal(value)))
-        } else if let Some(tt) = ast.token_tree() {
-            Some(Interned::new(AttrInput::TokenTree(ast_to_token_tree(&tt).0)))
```
Comment (matklad) [inline crates/hir_def/src/attr.rs:679]: The old version looks more readable to me, the two cases can be symmetric.

File: crates/mbe/src/expander/matcher.rs [hunk 2]
```diff
@@ -756,19 +756,19 @@ impl<'a> TtIter<'a> {
         let ok = match separator {
             Separator::Ident(lhs) if idx == 0 => match fork.expect_ident_or_underscore() {
                 Ok(rhs) => rhs.text == lhs.text,
-                _ => false,
+                Err(_) => false,
```
Comment (matklad) [inline crates/mbe/src/expander/matcher.rs:759]: :+1:

File: crates/mbe/src/lib.rs [hunk 3]
```diff
@@ -268,7 +268,7 @@ impl MacroDef {
 }
 
 impl Rule {
-    fn parse(src: &mut TtIter, expect_arrow: bool) -> Result<Rule, ParseError> {
+    fn parse(src: &mut TtIter, expect_arrow: bool) -> Result<Self, ParseError> {
```
Comment (matklad) [inline crates/mbe/src/lib.rs:271]: The original version looks more readable to me.

File: crates/hir/src/lib.rs [hunk 4]
```diff
@@ -992,9 +991,7 @@ impl Function {
     }
 
     pub fn method_params(self, db: &dyn HirDatabase) -> Option<Vec<Param>> {
-        if self.self_param(db).is_none() {
-            return None;
-        }
+        self.self_param(db)?;
```
[the remark this answers is not in the document]
Comment (clemenswasser) [reply crates/hir/src/lib.rs:994]: Though we also use it for Results (which is basically the same pattern as here).
The relevant clippy lint is [here](https://rust-lang.github.io/rust-clippy/master/index.html#question_mark)

File: crates/mbe/src/lib.rs (hunk 3 above)
Comment (clemenswasser) [reply crates/mbe/src/lib.rs:271]: Oh sorry, I forgot to revert this one

**Comment by matklad** [comment]
ping @clemenswasser , this needs a rebase and backing out of false positives :-)

**Review by matklad — COMMENTED** [review]
bors r+

Thanks!

**Pull request merged**